### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by bfmatei**
This PR is an all-in-one bringing multiple changes:
- docs
- added `markdownlint`
- moved the `console` transport into the console plugin
- the console plugin now ignores debug and log messages out of the box
- prevent endless loops while printing messages in the console
- moved the `fetch` transport into its own separate plugin
- updated dependencies

### CODE REVIEW

**Review by domasx2 — APPROVED** [review]
Looks great! 

But I'd consider some restructuring/renaming to make it clearer:
* what is for browsers and what is for node/deno(?)
* what is a transport and what is instrumentation

File: README.md [hunk 1]
```diff
@@ -1 +1,39 @@
...
+
+### Plugins
+
+Official plugins:
+
+#### Functionality
+
+1. [@grafana/javascript-agent-plugin-console](https://github.com/grafana/grafana-javascript-agent/tree/main/packages/plugin-console)
+   is a plugin for both, browsers and Node.js, which provides an automatic mechanism for collecting console events.
+1. [@grafana/javascript-agent-plugin-errors](https://github.com/grafana/grafana-javascript-agent/tree/main/packages/plugin-errors)
```
[the remark this answers is not in the document]
Comment (bfmatei) [reply README.md:28]: What do you think about having the same package for both and then just configure the plugin which environment he is on. Or better: have a helper in the `core` package that tells which env it is running.

File: packages/core/README.md [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+
+The entire architecture of the library is contained within this package. Out of the box, it doesn't collect any metrics,
+logs etc. but it offers an API to capture them.
+
+---
+
+## Agent
+
+The agent is an object which can be accessed by either importing it from the package or by referencing it from the
+global object (`window` in browsers and `global` in Node.js).
```
[the remark this answers is not in the document]
Comment (bfmatei) [reply packages/core/README.md:13]: Working on this one today. Basically it's `window.grafanaJavaScriptAgent.api.pushLog`

File: packages/plugin-console/src/index.ts [hunk 3]
```diff
@@ -1,35 +1,54 @@
...
-];
-
-export default function getPlugin(disabledLevels: LogLevel[] = []): Plugin {
-  return {
+const defaultDisabledLevels: LogLevel[] = [LogLevel.DEBUG, LogLevel.LOG];
+
+export interface ConsolePluginOptions {
+  disabledInstrumentationLevels?: LogLevel[];
+  enableInstrumentation?: boolean;
+  enableTransport?: boolean;
```
[the remark this answers is not in the document]
Comment (bfmatei) [reply packages/plugin-console/src/index.ts:21]: I was thinking about grouping similar functionalities into the same packages. Hence, a plugin that handles the console can contain both?

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (domasx2) [reply README.md:28]: If I was a user, I'd rather configure it myself explicitly :thinking:  perhaps autodetect with sane plugin defaults only if nothing is provided excplitly

**Review by domasx2 — DISMISSED** [review]
lgtm, great effort!

File: packages/core/README.md [hunk 4]
```diff
@@ -5,7 +5,134 @@ Core package of Grafana JavaScript Agent.
...
+
+The `api` property on the agent contains all the necessary methods to push new events.
+
+##### Exceptions
+
+- `pushException` - is a method to push an error/exception to the agent. It accepts a message as a mandatory parameter
+  and two optional ones: a `type` which by default is `error` and `stackFrames` which is an array of stack frames.
+
+  ```ts
+  agent.api.pushException('This is an error');
```
[the remark this answers is not in the document]
Comment (bfmatei) [reply packages/core/README.md:60]: For the time being, we don't support `Error` object, but we will.

File: packages/core/README.md [hunk 5]
```diff
@@ -5,7 +5,134 @@ Core package of Grafana JavaScript Agent.
 The entire architecture of the library is contained within this package. Out of the box, it doesn't collect any metrics,
 logs etc. but it offers an API to capture them.
 
-## Agent
+## Installation
+
+```ts
+import { initializeAgent } from '@grafana/javascript-agent-core';
+
+initializeAgent({
```
[the remark this answers is not in the document]
Comment (bfmatei) [reply packages/core/README.md:13]: Done

**Review by domasx2 — APPROVED** [review]

**Pull request merged**